compose.c

```diff
@@ -359,24 +372,151 @@ static void autocrypt_compose_menu(struct Email *e)
 #endif
 
 /**
- * redraw_crypt_lines - Update the encryption info in the compose window
- * @param rd Email and other compose data
+ * draw_floating - Draw a floating label
+ * @param win  Window to draw on
+ * @param row  Row to draw at
+ * @param col  Column to draw at
+ * @param text Text to display
  */
-static void redraw_crypt_lines(struct ComposeRedrawData *rd)
+static void draw_floating(struct MuttWindow *win, int row, int col, const char *text)
 {
-  struct Email *e = rd->email;
+  mutt_curses_set_color(MT_COLOR_COMPOSE_HEADER);
+  mutt_window_mvprintw(win, row, col, "%s", text);
+  mutt_curses_set_color(MT_COLOR_NORMAL);
+}
 
+/**
+ * draw_header - Draw an aligned label
+ * @param win   Window to draw on
+ * @param row   Row to draw at
+ * @param field Field to display, e.g. #HDR_FROM
+ */
+static void draw_header(struct MuttWindow *win, int row, enum HeaderField field)
+{
   mutt_curses_set_color(MT_COLOR_COMPOSE_HEADER);
-  mutt_window_mvprintw(rd->win, HDR_CRYPT, 0, "%*s", HeaderPadding[HDR_CRYPT],
-                       _(Prompts[HDR_CRYPT]));
+  mutt_window_mvprintw(win, row, 0, "%*s", HeaderPadding[field], _(Prompts[field]));
   mutt_curses_set_color(MT_COLOR_NORMAL);
+}
```

Comment:
We could possibly merge these two functions, with a size_t to indicate the width. We could use 0 for "no padding". `printf("%*s\n", 0, "Hello world");` is fine.

Reply:
Hmm... that would change eleven calls:
```diff
- draw_header(rd->win_envelope, row++, HDR_NEWSGROUPS);
+ draw_header(rd->win_envelope, row++, HeaderPadding[HDR_NEWSGROUPS], _(Prompts[HDR_NEWSGROUPS]));
```